examples/aes-ccm/src/ccm_mode.rs

```diff
@@ -0,0 +1,197 @@
+use std::convert::TryInto;
+use hacspec_lib::*;
+use hacspec_aes::*;
+
+fn format_func(a: &ByteSeq, n: &ByteSeq, p: &ByteSeq, t: u8, alen: u64, nlen: u8, plen: u64) -> ByteSeq {
+    let mut r = 0;
```

Comment:
These lead to rustc warnings because they are overriden without use. That's a hacspec issue. But you can work around it by assigning the "default" value here instead of 0 and then skip the corresponding assignment later. Like for `tmp` you an do `tmp = 10` and skip the `else` branch below.